### ISSUE TITLE
Tracing bug: begin/end idle events mixed with begin/end processing

### ISSUE THREAD

**Issue by lifflander** [DARMA-tasking/vt#703]
**Describe the bug**
We are seeing in the generated log files begin/end idle events inside a `beginProcessing`/`endProcessing`. This seems to be happening both on develop and beta.5.

**Comment by pnstickne** [comment]
@lifflander Is there an example log file? Not yet able to reproduce locally with verification of the direct log file contents.

2 = begin, 3 = end
14 = begin idle, 15 = end idle.

```
PROJECTIONS-RECORD 0
6 0
1 4 6 387 1 3 72 0
2 4 5 789 2 1 56 0 0 0 0 0 0
1 4 6 801 2 3 72 0
3 4 5 807 2 1 56 0 0 0 0 0 0
2 4 5 876 4 1 56 0 0 0 0 0 0
1 4 6 879 3 3 72 0
3 4 5 884 4 1 56 0 0 0 0 0 0
1 4 2 1396 4 3 80 0
1 4 21 1504 5 3 80 0
14 1513 3
15 1535 3
1 4 21 1568 6 3 80 0
14 1599 3
15 1629 3
2 4 0 1634 6 1 56 0 0 0 0 0 0
1 4 0 1747 7 3 80 0
3 4 0 1774 6 1 56 0 0 0 0 0 0
14 1826 3
15 1917 3
2 4 0 1942 9 1 56 0 0 0 0 0 0
1 4 0 1969 8 3 80 0
3 4 0 1979 9 1 56 0 0 0 0 0 0
14 2017 3
15 2070 3
2 4 3 2073 5 0 56 0 0 0 0 0 0
3 4 3 2105 5 0 56 0 0 0 0 0 0
14 2112 3
15 2320 3
1 4 21 2328 9 3 80 0
1 4 21 2363 10 3 80 0
14 2378 3
15 2409 3
1 4 21 2413 11 3 80 0
14 2427 3
15 2441 3
1 4 21 2445 12 3 80 0
14 2460 3
15 2502 3
1 4 21 2506 13 3 80 0
```

**Comment by pnstickne** [comment]
I can repro a mis-ordered event time:

So if when correctly ordered this would be a partial overlap.
```
2 4 6 39395 10 3 176 0 0 0 0 0 0
3 4 6 39395 10 3 176 0 0 0 0 0 0
14 39407 3
15 181161 3
2 4 49 181159 13 0 48 0 0 0 0 0 0
3 4 49 181250 13 0 48 0 0 0 0 0 0
```

**Comment by pnstickne** [comment]
The PR fixes the mis-ordered time (identified above) and it _might_ fix issues that look like the following which I suspect is from an "empty" work queue when the scheduler is run recursively. 

However, *the changes on the PR might actually HIDE idle time* (such as if a good bit of time is added in a barrier wait) - it really depends on what idle means.

```
2 ..
14 ..
15 ..
3 ..
```

**Comment by pnstickne** [comment]
Indeed, confirmed with barriers (and would affect other usages):

> 2 4 0 5487 18 4 48 0 0 0 0 0 0
> 14 5488 1
> 3 4 0 5489 18 4 48 0 0 0 0 0 0
> 2 4 0 5489 18 3 48 0 0 0 0 0 0
> 3 4 0 5491 18 3 48 0 0 0 0 0 0
> 2 4 0 5491 19 2 48 0 0 0 0 0 0
> 3 4 0 5492 19 2 48 0 0 0 0 0 0
> 2 4 0 5492 19 0 48 0 0 0 0 0 0
> 3 4 0 5493 19 0 48 0 0 0 0 0 0
> 15 5517 1
> 2 4 3 5517 24 0 48 0 0 0 0 0 0
> 3 4 3 5519 24 0 48 0 0 0 0 0 0
> 14 5520 1
> 15 5524 1
> 20 4 0 5524 28 1 72 0 5

**Comment by pnstickne** [comment]
It seems like it is VALID to record the idle from these "nested scheduler runs" as well as the outside the schedule - that is, idle events CAN be nested as-a-result-of scheduling while running an action. In such a case it would be expected to see idle times under barrier/term-wait and similar cases.

However, currently the inner idle times are not guaranteed to be honest, which should be corrected..

**Comment by pnstickne** [comment]
Or, as @lifflander tells me that's a tool limitation.. explicitly (and automatically) wrapping the nested calls in a begin/end event pair.

**Comment by PhilMiller** [comment]
This claims to have been addressed by #733, which was indeed merged in beta 9
